### PR TITLE
Upgrade jwt/bundler to more modern dependencies.

### PR DESCRIPTION
**Pull request by patricksrobertson**
JWT 1.4 is four years old. The encode call seems to be the same between 1.4 and 2.2.

Locking to bundler 1.6 is also somewhat old and out of date.

### CODE REVIEW

**Comment by xhocquet** [comment]
Hey @patricksrobertson , thanks for the contribution! 🌟 

Bundler seems fine. What do you think about removing the versioning dependency on JWT as well? My thought is that users should use the most recent version in their Gemfile, and if they pull a buggy patch as a requirement somewhere else in their application then they should notice failures beyond just Boxr.

**Comment by xhocquet** [comment]
Actually, thinking about this a little more:

For Bundler, removing the versioning means v2.0 is allowed, which removes support for Ruby versions <2.3. This gem supports Ruby >2.0, so that change is not acceptable.

Even for JWT, I don't see a reason to _require_ bumping the version. While I see no breaking changes, users can already upgrade JWT to 2+ since it passes the version requirements of this gem.

In short: Bundler change could break backwards compatibility. JWT change is unnecessary. Closing for now, feel free to re-open if you think otherwise!

**Comment by patricksrobertson** [comment]
I can’t use your gem for Box on my production app without relaxing these
requirements so I would love to find a way forward.

**Comment by xhocquet** [comment]
@patricksrobertson Could you please describe the other dependencies that are preventing you from using this library? I definitely want to help!

**Comment by patricksrobertson** [comment]
We either need to remove the jwt version requirement or lock it to the 2nd
major version. Bundled is a dev dependency and I can work around it.

**Comment by xhocquet** [comment]
@patricksrobertson What version of JWT does not satisfy the gemspec for Boxr? Based on my understanding, you can use JWT 2.0+ with the following requirements:
```
spec.add_runtime_dependency "jwt", ">= 1.4"
```

Is that not the case? Seems that way from the docs: https://guides.rubygems.org/patterns/#pessimistic-version-constraint

Example:
`gem 'library', '>= 2.2.0'`
> This is an “optimistic” version constraint. It’s saying that all versions greater than or equal to 2.2.0 will work with your software.
